### Problem Statement 1

Blood glucose levels for obese patients have a mean of 100 with a standard deviation of
15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.

In [1]:
import numpy as np
from scipy.stats import norm
import math

In [2]:
p_mean = 100
p_std = 15
n = 36
s_mean = 108
alpha = 0.05
SE = p_std/(n)**0.5
z_stat = (s_mean - p_mean)/SE
print("z_stat is: ", z_stat)
print("Critical region is : ", norm.ppf(alpha/2), "and", -norm.ppf(alpha/2))

z_stat is:  3.2
Critical region is :  -1.9599639845400545 and 1.9599639845400545


### Null hypothesis is rejected as Z_stat lies in the critical region at 5% level of significance. So, raw cornstarch has an effect on blood glucose level.

### Problem statement 2

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?

In [4]:
n1 = 100
n2 = 100
R1 = 0.52            #Republicans from state 1
D1 = 0.48            #Democrats from state 1
R2 = 0.47            #Republicans from state 2
D2 = 0.53            #Democrats from state 2

# Probability of R1 > R2
std = Standard_deviation = math.sqrt((R1*D1)/n1 + (R2*D2)/n2)
# Mean difference of the number of republicans in both states = 0.52 - 0.47 = 0.05
mu = 0.05
# Difference should be less than 0 for greater percentage of Republican voters in the second  state than in the first state
# Hence, x = 0
x = 0
z_score = (x - mu)/std
z_score = (0 - 0.05)/std
print("z_score is: ", z_score)

z_score is:  -0.7080278164104207


### From Z score table -0.708 refers to 0.24
### Hence 0.24 is the probability that the survey will show a greater percentage of Republican voters in the second  state than in the first state



### Problem statement 3

You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?

In [5]:
x = 1100
mu = 1026
std = 209
z_score = (x - mu)/std
print("z_score is: ", z_score)

z_score is:  0.35406698564593303


In [6]:
print("Range of first standard deviation: ", mu-std, "and", mu+std)

Range of first standard deviation:  817 and 1235


### z_score value shows that my score is 0.35 standard deviation above mean. It lies in the first standard deviation range. So, performance is quite good.